bitcoin/src/blockdata/script/borrowed.rs

```diff
@@ -483,14 +483,16 @@ crate::internal_macros::define_extension_trait! {
         /// Writes the human-readable assembly representation of the script to the formatter.
         #[deprecated(since = "TBD", note = "use the script's Display impl instead")]
         fn fmt_asm(&self, f: &mut dyn fmt::Write) -> fmt::Result {
-            bytes_to_asm_fmt(self.as_ref(), f)
+            write!(f, "{}", self)
         }
 
         /// Returns the human-readable assembly representation of the script.
         #[deprecated(since = "TBD", note = "use `to_string()` instead")]
         fn to_asm_string(&self) -> String {
+            use core::fmt::Write;
+
             let mut buf = String::new();
-            bytes_to_asm_fmt(self.as_ref(), &mut buf).expect("in-memory writers don't fail");
+            write!(&mut buf, "{}", self).expect("in-memory writers don't err");
```

Comment:
In 84af64671aa72cc39f5bf67546496d827469601b:

It's fine to unwrap here but the text in the expect message makes no sense. The reason is that "the `fmt::Display` impl for `Self` does not return an error". It has nothing to do with memory or writers.

Also, the commit message says you don't know why this method used to exist but you think it is from when Script was Sized. It has nothing to do with Sized. It is because the `Display` impl used to add a `Script(...)` wrapper to its output and users (Steven) wanted to get the text out without that noise.

Reply:
Oh I thought the `Write` trait has an error for I/O failures but writes to a `String` (eg writing to memory) never fail.

Thanks for clarifying the other thing, will fix.

Reply:
I'd like to clear this up because if I see this line in code `write!(&mut buf, "{}", self).unwrap();` I have to ask myself "why is it ok to unwrap?", so I think using `expect("exactly correct reason why this won't panic")` is better.

Reply:
In this case, probably the best thing to write is "if this returns an error, `to_string` will panic, which would be a bug in its own right".

Reply:
Basically, I don't think you're really justified in having this unwrap here...but the implementation of `ToString` on `T: Display` in core has the same unwrap, and so everybody who implements `fmt::Display` has responded by bending over backward to never actually return a `fmt::Result::Err`, so we're allowed to pretend like the format functions are infallible.

Reply:
Cheers, turns out I'm a wombat anyways, we can just call `self.to_string()` and bypass the whole issue - face palm.